File: backend/src/models/ApiRequest.js
const mongoose = require('mongoose');

const ApiRequestSchema = new mongoose.Schema({
  name: { type: String, required: true },
  method: { 
    type: String, 
    required: true, 
    enum: ['GET', 'POST', 'PUT', 'PATCH', 'DELETE', 'HEAD', 'OPTIONS'] 
  },
  url: { type: String, required: true },
  headers: [{ key: String, value: String, enabled: Boolean }],
  params: [{ key: String, value: String, enabled: Boolean }],
  body: {
    type: { type: String, enum: ['none', 'json', 'text', 'form-data'], default: 'none' },
    content: mongoose.Schema.Types.Mixed // JSON string or object
  },
  collectionId: { type: mongoose.Schema.Types.ObjectId, ref: 'Collection' },
  folderId: String // Optional ID if nested in a folder
}, { timestamps: true });

module.exports = mongoose.model('ApiRequest', ApiRequestSchema);

File: inspecto/lib/domain/entities/api_request.dart
enum HttpMethod { GET, POST, PUT, DELETE, PATCH }

class ApiRequest {
  final String id;
  final HttpMethod method;
  final String url;
  final Map<String, String> headers;
  final Map<String, String> params;
  final String? body;
  final String? bodyType; // 'none', 'json', 'form'

  ApiRequest({
    required this.id,
    required this.method,
    required this.url,
    required this.headers,
    required this.params,
    this.body,
    this.bodyType = 'none',
  });
}
